src/app/ad-authoring/ad-authoring.state.ts

```diff
@@ -9,15 +9,177 @@ export interface AdAuthoringWorkflowState {
   readonly callToAction?: CallToActionEnum;
   readonly fileSrc?: string;
   readonly file?: File;
+  readonly AmpHtml: string;
 }
-// add a default state for the fields above
+
 @Injectable({
   providedIn: 'root',
 })
 export class AdAuthoringWorkflowStateContainer {
+  generateAMPHTML(
+    callToActionStr: string,
+    landingUrl: string,
+    landingType: string,
+    base64AssetStr: string,
+    assetFile: File
+  ) {
+    const adHtml =
+      '"<!doctype html><html amp4ads><head><meta charset=\\"utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width,minimum-scale=1\\"><meta name=\\"amp-cta-type\\" content=\\"' +
+      callToActionStr +
+      '\\"><meta name=\\"amp-cta-url\\" content=\\"' +
+      landingUrl +
+      '\\"><meta name=\\"amp-cta-landing-page-type\\" content=\\"' +
+      landingType +
+      '\\"><style amp4ads-boilerplate>body{visibility:hidden}<\\/style><script async src=\\"https:\\/\\/cdn.ampproject.org\\/amp4ads-v0.js\\"><\\/script>';
+
+    const ampVideoRuntime =
+      '<script async custom-element=\\"amp-video\\" src=\\"https:\\/\\/cdn.ampproject.org\\/v0\\/amp-video-0.1.js\\"><\\/script>';
+
+    const imageHtml =
+      '<\\/head><body><amp-img layout=\\"fixed\\" height=\\"250\\" width=\\"300\\" src=\\"' +
```

Comment:
Is there a way we could identify the size of an image from the upload? or should we provide input options?

Reply:
Will address this in a separate PR